src/components/NewProject/index.jsx

```diff
@@ -21,12 +27,51 @@ function NewProject() {
   const code = query.get("code");
   const state = query.get("state");
 
+  const updateFileKey = newFileKey => {
+    usePageStatusStore.getState().setStatus({ fileKey: newFileKey });
+  };
+
+  const updateVersions = newVersions => {
+    useProjectVersionStore.getState().setVersion(newVersions);
+  };
+
+  const handleModalClick = ev => {
+    ev.preventDefault();
+
+    setIsModalOpened(false);
+  };
+
+  const handleInputChange = ev => {
+    setInputValue(ev.target.value);
+  };
+
+  const onClickSubmitButton = async ev => {
+    ev.preventDefault();
+
+    const fileKey = getFileKeyFromURI(inputValue);
+
+    updateFileKey(fileKey);
+    updateVersions(await getVersions(fileKey));
+
+    navigate("/version");
+  };
+
+  const setAuthStore = authParams => {
+    useAuthStore.getState().setAuth(authParams);
+  };
```

Comment:
P2: 관련 함수 로직 간 묶어주는 게 어떨까요? `Store`에 저장하는 함수 로직과 이벤트 핸들러 함수 로직의 순서를 
같이 묶어주면 더 좋을 것 같습니다 👍 

```suggestion
  const setAuthStore = authParams => {
    useAuthStore.getState().setAuth(authParams);
  };

  const updateFileKey = newFileKey => {
    usePageStatusStore.getState().setStatus({ fileKey: newFileKey });
  };

  const updateVersions = newVersions => {
    useProjectVersionStore.getState().setVersion(newVersions);
  };

  const handleModalClick = ev => {
    ev.preventDefault();

    setIsModalOpened(false);
  };

  const handleInputChange = ev => {
    setInputValue(ev.target.value);
  };

  const onClickSubmitButton = async ev => {
    ev.preventDefault();

    const fileKey = getFileKeyFromURI(inputValue);

    updateFileKey(fileKey);
    updateVersions(await getVersions(fileKey));

    navigate("/version");
  };
```